invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/Documents/DocumentsDetails/DocumentItems/DocumentItems.js

```diff
@@ -0,0 +1,247 @@
+          activeLibrary: library,
+          activeInternalLocation: null,
+        };
+      } else {
+        this.state = {
+          activeLibrary: null,
+          activeInternalLocation: null
+        }
+      }
+    }
+  }
+
+  handleLocationClick = (e, titleProps) => {
+    const {name} = titleProps;
+
+    if (this.state.activeInternalLocation === name) {
+      this.setState({activeInternalLocation: null});
+    } else {
+      this.setState({activeInternalLocation: name});
+    }
+  };
+
+  handleClick = (e, titleProps) => {
+    const {name} = titleProps;
+
+    if (this.state.activeLibrary === name) {
+      this.setState({activeLibrary: null});
+    } else {
+      this.setState({activeLibrary: name});
+    }
+  };
+
+  renderInternalLocations = (internalLocations) => {
+    const locations = [];
+    const {activeInternalLocation} = this.state;
+
+    if(Object.keys(internalLocations).length > 2) {
+      Object.entries(internalLocations).forEach(
+        ([internalLocationName, items]) => {
+          if (internalLocationName !== 'total') {
+            locations.push(
+              <List.Item
+                active={activeInternalLocation === internalLocationName}
+                name={internalLocationName}
+                onClick={this.handleLocationClick}
+                key={internalLocationName}
+              >
+                {<Checkbox
+                  checked={activeInternalLocation === internalLocationName}/>}
+                {' '}
+                {internalLocationName}
+              </List.Item>
+            );
+          }
+        });
+      return <List className={'document-items-location-filters'}
+                   vertical>{locations}</List>;
+    }
+    return null;
+  };
+
+  renderLibraries = () => {
+    const {activeLibrary} = this.state;
+    const onShelf = this.props.documentDetails.metadata.items.on_shelf;
+    const libraries = [];
+
+    Object.entries(onShelf).forEach(
+      ([locationName, locationObj]) => {
+        libraries.push(
+          <Menu.Item key={locationName}>
+            <Accordion.Title
+              active={activeLibrary === locationName}
+              onClick={this.handleClick}
+              name={locationName}
+              content={
+                <><Label>{onShelf[locationName]["total"]}</Label>
+
+                  {locationName}</>
+              }
+            />
+            {Object.keys(locationObj).length > 2 ?
+              <Accordion.Content
+              active={activeLibrary === locationName}
+              content={this.renderInternalLocations(locationObj)}/>
+              : null
+            }
+          </Menu.Item>
+        );
+      });
+
+    return libraries;
+  };
+
+  renderItems = () => {
+    const onShelf = this.props.documentDetails.metadata.items.on_shelf;
+    if (!isEmpty(onShelf)) {
+      let items = [];
+      if (!this.state.activeLibrary) {
+        return "Click on an available location to see the items"
```

Comment:
IMHO, `items` is a technical term that we use in our domain. Shouldn't we display something like `copies/physical copies/available copies`?